src/Psd/PsdNativeFile_Linux.cpp

```diff
@@ -0,0 +1,164 @@
+
+#include "PsdPch.h"
+
+#include "PsdLog.h"
+#include "PsdFile.h"
+#include "PsdAllocator.h"
+#include "PsdNamespace.h"
+#include "PsdMemoryUtil.h"
+#include "PsdNativeFile_Linux.h"
+
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <unistd.h>
+#include <fcntl.h>
+#include <aio.h>
+
+#include <cstring>
+#include <cstdlib>
+#include <cwchar>
+#include <string>
+
+
+
+PSD_NAMESPACE_BEGIN
+/**
+ * @brief Convert wchar_t * to char *
+ * 
+ * @param ws 
+ * @return char * 
+ */
+static char *convert_ws(const wchar_t *ws){
+    char *buffer;
+    size_t n = std::wcslen(ws) * 4 + 1;
+    buffer = static_cast<char*>(std::malloc(n));
+    std::memset(buffer,0,n);
+    if(buffer == nullptr){
+        return nullptr;
+    }
+    std::wcstombs(buffer,ws,n);
+    return buffer;
+}
+
+NativeFile::NativeFile(Allocator *alloc):
+    File(alloc),
+    fd(-1){}
+
+//Convert wchar to char and open
+
+bool NativeFile::DoOpenRead(const wchar_t* filename){
+    char *name = convert_ws(filename);
+    fd = open(name,O_RDONLY);
+    if(fd == -1){
+        PSD_ERROR("NativeFile","open(%s) => %s",name,strerror(errno));
+        std::free(name);
+        return false;
+    }
+    std::free(name);
+    return true;
+}
```

Comment:
Do you see any reason why we cannot use the allocator? I'd prefer that to plain malloc and free.